### PR TITLE
IV-3181 - add TSS custom plugin

### PR DESCRIPTION
**Pull request by akarpik**

### CODE REVIEW

**Comment by raphael** [comment]
LGTM

File: rl10/design/resources/tss.rb [hunk 1]
```diff
@@ -25,6 +25,55 @@ class TSS
...
+    include Praxis::ResourceDefinition
+
+    description 'TSS Custom Plugins'
+    media_type 'text/plain'
+
+    routing do
+      prefix '/rll/tss'
+    end
+
+    action :get_exec do
```
Comment (douglaswth) [inline rl10/design/resources/tss.rb]: This action should be called `:index`.

File: rl10/design/resources/tss.rb [hunk 2]
```diff
@@ -25,6 +25,55 @@ class TSS
...
+      description 'Get plugins list'
+      routing { get '/exec' }
+      response :ok
+    end
+
+    action :post_exec do
```
Comment (douglaswth) [inline rl10/design/resources/tss.rb]: This action should be called `:create`.

File: rl10/design/resources/tss.rb [hunk 3]
```diff
@@ -25,6 +25,55 @@ class TSS
...
+      description 'Add new TSS custom pligin'
```
Comment (douglaswth) [inline rl10/design/resources/tss.rb]: Typo: `pligin` should be `plugin`.

File: rl10/design/resources/tss.rb [hunk 4]
```diff
@@ -25,6 +25,55 @@ class TSS
...
+      routing { post '/exec' }
+      params do
+        attribute :name, String, required: true
+        attribute :executable, Attributor::Collection.of(String), required: true
+      end
+      response :ok
+    end
+
+    action :get_exec_name do
```
Comment (douglaswth) [inline rl10/design/resources/tss.rb]: This should be called `:show`.

File: rl10/design/resources/tss.rb [hunk 5]
```diff
@@ -25,6 +25,55 @@ class TSS
...
+      description 'Get TSS plugin info'
+      routing { get '/exec/:name' }
+      response :ok
+    end
+
+    action :put_exec_name do
```
Comment (douglaswth) [inline rl10/design/resources/tss.rb]: This should be called `:update`.

File: rl10/design/resources/tss.rb [hunk 6]
```diff
@@ -25,6 +25,55 @@ class TSS
...
+      description 'Update TSS custom pligin'
+      routing { put '/exec/:name' }
+      params do
+        attribute :executable, Attributor::Collection.of(String), required: true
+      end
+      response :ok
+    end
+
+    action :delete_exec_name do
```
Comment (douglaswth) [inline rl10/design/resources/tss.rb]: This should be called `:destroy`.

File: rl10/design/resources/tss.rb [hunk 7]
```diff
@@ -25,6 +25,55 @@ class TSS
...
+    end
+
+    action :show do
+      description 'Get TSS plugin info'
+      routing { get '/exec/:name' }
+      response :ok
+    end
+
+    action :update do
+      description 'Update TSS custom pligin'
```
Comment (douglaswth) [inline rl10/design/resources/tss.rb]: Hmm, looks like this typo is still here: `pligin` should be `plugin`.

**Pull request merged**